sklearn/linear_model/base.py

```diff
@@ -56,12 +57,19 @@ def make_dataset(X, y, sample_weight, random_state=None):
     # seed should never be 0 in SequentialDataset
     seed = rng.randint(1, np.iinfo(np.int32).max)
 
+    if X.dtype == np.float32:
+        CSRData = CSRDataset32
+        ArrayData = ArrayDataset32
```

Comment:
Currently, there isn't any specific test for ``make_dataset``. It's supposed to be covered by the tests for the sag solver. So we can either put the test off to a next PR about the sag solver (#9020) or doing the test you are proposing in this PR. What do you prefer?